src/Microsoft.ML.Data/Evaluators/MulticlassClassificationEvaluator.cs

```diff
@@ -172,6 +172,7 @@ public override IEnumerable<MetricColumn> GetOverallMetricColumns()
                     logLossRed.Add(agg.UnweightedCounters.Reduction);
                     if (agg.UnweightedCounters.OutputTopKAcc > 0)
                         topKAcc.Add(agg.UnweightedCounters.TopKAccuracy);
+                    allTopK.Add(agg.UnweightedCounters.AllTopKAccuracy);
```

Comment:
If we go to limit the size of the `TopKAccuracyForAllK` array (as suggested here), then we'd actually want this inside the if-statement above (i.e. only add to `alltopK` List if `agg.UnweightedCounters.OutputTopKAcc > 0 `)... the same should happen for the `agg.WeightedCounter...` branch